src/main/cpp/src/slice.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/lists/lists_column_view.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+namespace spark_rapids_jni {
+
+std::unique_ptr<cudf::column> slice(
```

Comment:
Better to have comment to desribe these APIs on its functionality and input and output, along with some simple examples.
At least, we should explain these APIs are used to slice each row of a column of list, not like the exsiting `slice` in cudf to slice the entire column.